app/scenes/Document/components/SharePopover.tsx

```diff
@@ -233,6 +266,17 @@ function SharePopover({
               </SwitchText>
             </SwitchLabel>
           </SwitchWrapper>
+          <Separator />
+          <SwitchWrapper>
+            <Input
+              type="text"
+              label={t("Custom link")}
+              onChange={handleUrlSlugChange}
+              error={slugValidationError}
+              preview={urlSlug ? `${team.url}/${urlSlug}` : ""}
+              defaultValue={urlSlug || ""}
```

Comment:
Why do we need `urlSlug` in state if `share.urlId` is already observable?

Well, consider the scenario where someone types "foobar*", an invalid value in the `input` and then closes the share popover. Now, if the popover is opened again, the expected value in the input should be "foobar*", which is stored in `urlSlug`, not `share.urlId`.